### PR TITLE
Endianness fixes

### PR DESCRIPTION
**Pull request by tbzatek**
One more change for #4

### CODE REVIEW

File: src/nvme/nbft.c [hunk 1]
```diff
@@ -96,15 +99,17 @@ static int __get_heap_obj(struct nbft_header *header, const char *filename,
 	}
 
 	/* check that string is zero terminated correctly */
-	*output = (char *)header + obj.offset;
+	*output = (char *)header + le32_to_cpu(obj.offset);
```
Comment (johnmeneghini) [inline src/nvme/nbft.c:102]: These are all major bug fixes.  How did this stuff work before these changes?

**Review by johnmeneghini — APPROVED** [review]
These are all good changes.  I just don't understand how any of this worked and I thought we wanted to remove the ccan dependencies.

File: src/nvme/nbft.c [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include <stdlib.h>
 
 #include <arpa/inet.h>
+#include <ccan/endian/endian.h>
```
[the remark this answers is not in the document]
Comment (tbzatek) [reply src/nvme/nbft.c:15]: From public headers only, `ccan` is still widely used internally and linked statically.

File: src/nvme/nbft.c (hunk 1 above)
Comment (tbzatek) [reply src/nvme/nbft.c:102]: I suppose big endian architectures are pretty rare nowadays, this still needed to be done for the sake of completeness. Same philosophical question goes for 32-bit archs.

**Pull request merged**